File: Swift/shortest-distance-to-a-character.swift
class Solution {
    func shortestToChar(_ S: String, _ C: Character) -> [Int] {
        let chars = Array(S)
        let count = S.count
        var res = Array(repeating: 0, count: count)
        var currentPosition = -count

        for i in 0..<count {
            if chars[i] == C {
                currentPosition = i
            }
            res[i] = i - currentPosition
        }

        for i in Array(0..<count).reversed() {
            if chars[i] == C {
                currentPosition = i
            }
            res[i] = min(res[i], abs(currentPosition - i))
        }

        return res
    }
}

File: Python/shortest-distance-to-a-character.py
class Solution:
    def shortestToChar(self, S, C):
        """
        :type S: str
        :type C: str
        :rtype: List[int]
        """
        length = len(S)
        res, currentPosition = [0] * length, -length

        for i in range(length):
            if S[i] == C:
                currentPosition = i
            res[i] = i - currentPosition

        for i in range(length - 1, -1, -1):
            if S[i] == C:
                currentPosition = i
            res[i] = min(res[i], abs(currentPosition - i))

        return res
